devices/debian.py

```diff
@@ -55,9 +55,8 @@ def __init__(self,
             raise Exception("Unable to connect to %s." % name)
         if i == 0:
             self.sendline("yes")
-            self.expect("assword:")
-            self.sendline(password)
-        elif i == 1:
+            i = self.expect(["Last login", "assword:"])
+        if i == 1:
```

Comment:
Because I deleted self.sendline(password) :-) Now the if statement on success fills i with 1 if password is needed and 0 if not and therefore the next if is a valid check whether to send a password or not.